File: 238-product-of-array-except-self/238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int c=0,s=1;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0) c++;
            else s=s*nums[i];
        }
        int a[]=new int[nums.length];
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                if(c>1){
                    a[i]=0;
                }
                else{
                    a[i]=s;
                }
            }
            else{
                if(c>0)
                    a[i]=0;
                else{
                    a[i]=s/nums[i];
                }
            }
        }
        return a;
    }
}

File: 238-product-of-array-except-self/238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int count=0,p=1;
        vector<int> v;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                count++;
            } 
            else p=p*nums[i];
            
        }
        if(count>1) {
            for(int i=0;i<nums.size();i++){
                v.push_back(0);
            }
        }
        else if(count==1){
            for(int i=0;i<nums.size();i++){
                if(nums[i]!=0) v.push_back(0);
                else v.push_back(p);
            }
        }
        else{
            for(int i=0;i<nums.size();i++){
                v.push_back(p/nums[i]);
            }
        }
        return v;
    }
};